lib/realtime/server.ex

```diff
@@ -187,17 +213,20 @@ defmodule Realtime.Server do
   @spec subscribe(GenServer.server(), {:run_ids, [Run.id()]}) :: [VehicleOrGhost.t()]
   @spec subscribe(GenServer.server(), {:block_ids, [Block.id()]}) :: [VehicleOrGhost.t()]
   @spec subscribe(GenServer.server(), {:alerts, Route.id()}) :: [String.t()]
+  defp subscribe(server, {:alerts, _route_id} = subscription_key) do
+    {pubsub, ets} = GenServer.call(server, :subscription_info)
+    Phoenix.PubSub.subscribe(pubsub, "realtime_alerts")
+    lookup({ets, subscription_key})
+  end
+
   defp subscribe(server, subscription_key) do
-    {registry_key, ets} = GenServer.call(server, :subscription_info)
-    Registry.register(Realtime.Registry, registry_key, subscription_key)
+    {pubsub, ets} = GenServer.call(server, :subscription_info)
+    Phoenix.PubSub.subscribe(pubsub, "realtime_vehicles")
     lookup({ets, subscription_key})
   end
 
   defp update_subscription(server, {:limited_search, _search_params} = subscription_key) do
```

Comment:
question (non-blocking): Is update_subscription still needed?

Reply:
The main reason I decided to keep it around is so that `lookup` actually gets called immediately when updating the subscription, rather than having to wait for the next data update (which is usually going to take only around a second, but I figured this way it's more responsive to user input).

Reply:
That makes sense! Maybe a rename would be helpful for this function since it is no longer updating a subscription, though still non-blocking!

Reply:
I'm in more-of-less the same boat, I think it needs a new name but I can't determine a good one. `get_latest_results` is probably better than anything I've thought up, though.